cuappdev/resell-react-native
# Integrate AppDev Announcements

## Overview

Add AppDev Announcements by integrating with the `react-native-appdev-announcements` package.

## Changes Made

- Add the `AppDevAnnouncements` component and passed in the `host` and `appPath` which are provided through environment variables.
- **Note that an updated .env file is needed (already edited on Slack).**
- Add `pull_request_template.md` for PR templates.
- Add `react-native-svg` to `"resolutions"` in `package.json` due to conflicting versions from other packages.

## Test Coverage

To test this PR, do the following:
1. Make sure all dependencies are updated with `yarn`.
2. Use the new `.env` files containing `ANNOUNCEMENTS_PATH` and `ANNOUNCEMENTS_HOST` environment variables.
3. Inside of a `useEffect` in `HomeScreen.tsx`, call the `resetAnnouncements()` function provided by the `react-native-appdev-announcements` package.
4. When the app launches, there should be an announcement popup within 4 seconds. This banner will dismiss after 8 seconds, with animation.

## Related PRs or Issues

#210 and #207 are rebased in this PR. Upon merge of this PR, the related PRs should be closed.

## Next Steps
Update our CI/CD pipeline on EAS with the new environment variables.

## Screenshots


<details>
  <summary>Screenshot</summary>
  <img width="384" alt="Screenshot 2024-09-09 at 2 58 16 PM" src="https://github.com/user-attachments/assets/33d61045-acaf-43e2-a8d8-0a6abaac5a4e"></details>
</details>

<details>
  <summary>Video (note the image not showing is due to my poor connection)</summary>
<video src="https://github.com/user-attachments/assets/c50f13dc-a649-4372-8a5f-dc8ea0c970be
" type="video/mp4">
</details>
